Update component-base owners to include SIG-Architecture

mtaufen opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Update component-base owners to include SIG-Architecture, as a stakeholder SIG of the (soon-to-be-dissolved) WG Component Standard, and as the owners of code organization projects.

#### Special notes for your reviewer:

@neolit123 suggested that SIG-Cluster-Lifecycle doesn't have much of a stake in component-base, maybe we remove the sig/cluster-lifecycle label in this PR also?

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

SIG-Architecture approvers:
/assign @derekwaynecarr @dims @johnbelamaric

mtaufen committed:
Update component-base OWNERS to include SIG-Architecture

Also update OWNERS in subdirs based on best-guess picks of appropriate
OWNERS. We can update as necessary in future iterations but this'll help
reduce initial noise for SIG-Arch.

dims commented on staging/src/k8s.io/component-base/OWNERS:
```diff
@@ -1,12 +1,14 @@
 # See the OWNERS docs at https://go.k8s.io/owners
 
 approvers:
+- sig-architecture-leads
```
please use `sig-architecture-approvers` from root OWNERS_ALIASES

dims commented on staging/src/k8s.io/component-base/OWNERS:
```diff
@@ -1,12 +1,14 @@
 - sttts
 - lavalamp
 reviewers:
+- sig-architecture-pr-reviews
```
you can drop this line

mtaufen replied:
double-checking; with @neolit123's suggestion this be the only reviewer line, do we still want to drop?

mtaufen replied:
I guess the bot is complaining about org membership for this identity... alternative?

neolit123 replied:
i know that there is a bug in test-infra owner assignment code / bots, where if there are no listed reviewers it would not take people from the list of approvers, but rather take owners from the parent OWNERS; in this case that would be:
https://github.com/kubernetes/kubernetes/blob/master/staging/OWNERS

mtaufen replied:
Ok, went with sig-architecture-approvers for both of these for now.

dims commented on staging/src/k8s.io/component-base/OWNERS:
```diff
@@ -16,5 +18,6 @@ reviewers:
 emeritus_approvers:
 - jbeda
 labels:
+- sig/architecture
```
sure!

neolit123 commented:
as mentioned on the ML thread i think this should just have sig-arch at the root to gate packages additions and removals.

e.g.
```
approvers:
- sig-architecture-leads
reviewers:
- sig-architecture-pr-reviews
labels:
- sig/architecture
```

`luxas`, `dixudx`, `rosti`, `stewart-yu` haven't been active in k8s for a while so i think we should move them to emeritus / remove for the time being (great if they come back at some point...).

at least sig-instrumentation have correctly set their ownership in component-base packages.

ideally, we need to figure out owning sigs for these packages:
https://github.com/kubernetes/kubernetes/tree/39e2133619ccd48f7c7b68c9b96ba9c8fa46af5e/staging/src/k8s.io/component-base/cli (CLI?)
https://github.com/kubernetes/kubernetes/tree/39e2133619ccd48f7c7b68c9b96ba9c8fa46af5e/staging/src/k8s.io/component-base/codec (strict codec, api-machinery)?
https://github.com/kubernetes/kubernetes/tree/39e2133619ccd48f7c7b68c9b96ba9c8fa46af5e/staging/src/k8s.io/component-base/config (this has LeaderElectionConfiguration used in scheduler and KCM, sig-scheduling + api-machinery?)
https://github.com/kubernetes/kubernetes/tree/39e2133619ccd48f7c7b68c9b96ba9c8fa46af5e/staging/src/k8s.io/component-base/configz (this is code from 2015 that was used in KCM, CCM, kube-proxy etc for enabling serving a componentconfig at a /configz endpoint, unclear owner, maybe sig-arch, or sig-cluster-lifecycle given componentconfig originated there?).
https://github.com/kubernetes/kubernetes/tree/39e2133619ccd48f7c7b68c9b96ba9c8fa46af5e/staging/src/k8s.io/component-base/featuregate (who owns feature gates as a mechanism? sig-arch?)
https://github.com/kubernetes/kubernetes/blob/39e2133619ccd48f7c7b68c9b96ba9c8fa46af5e/staging/src/k8s.io/component-base/term/term.go (used in various components as a utility; imports a moby package, unclear owner)
https://github.com/kubernetes/kubernetes/tree/39e2133619ccd48f7c7b68c9b96ba9c8fa46af5e/staging/src/k8s.io/component-base/version (used by most k/k produced binaries to pin versions, sig-arch?, sig-release?)

we had similar problems in /cluster; just an umbrella owning sig + some owners and the sub-packages had code from various sigs - scalability, testing, etc.

mtaufen commented:
Thanks @neolit123. Updated to make sig-arch the root owner and moved individuals to emeritus. 

I'll take a pass at the rest of your suggestions later in this same PR. Might as well get a first approximation of those to reduce noise before we merge this.

mtaufen commented:
Took a first pass at sub-OWNERS, @dims @neolit123 PTAL.

neolit123 reviewed (commented):
Thanks, I agree with the owner updates. Not sure what others think.
/lgtm

mtaufen commented:
/assign @lavalamp

lavalamp commented:
Is there any overlap in approvers from before and after this change? I don't think it's good to break continuity completely if not.

mtaufen commented:
@lavalamp I suppose I can leave the two of us in the root approvers for a little while as a fallback... but given the others listed/the WG are not actively contributing in this area anymore I want the OWNERS files to reflect that.

lavalamp commented:
That's better than nothing. I don't think just giving it to SIG Arch is great if they're not invested in the code. Can someone link me to the mailing list conversation if this has already been debated?

lavalamp commented:
Note that I actually don't want to own this. But I don't see how it goes with SIG Arch. The most similar thing I can think of is the util packages, which sig owns those?

mtaufen commented:
Looks like I need to add you to the thread, sorry.

@dims had confirmed SIG-Arch as the backstop for code organization projects like this one.

lavalamp commented:
With the context there (specifically that SIG Arch is only for top-level changes), I guess I'm ok with this.

/approve
